From 9214435cc18f02d8aed6dd889d6428bb7e849f7d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Marchesin?= <marcheu@chromium.org>
Date: Tue, 17 Apr 2012 18:16:18 -0700
Subject: [PATCH 06/27] CHROMIUM: i965: Allow the case where multiple flush
 types are enqueued.

This happens when the miptree is allocated with intel_miptree_alloc_hiz
which adds NEED_HIZ_RESOLVE and then NEED_DEPTH_RESOLVE is added to it.

Conflicts:

	src/mesa/drivers/dri/i965/intel_resolve_map.c

Change-Id: Iec0095836d47dafb32034f459dbef6343f7058b0
---
 src/mesa/drivers/dri/i965/intel_resolve_map.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/mesa/drivers/dri/i965/intel_resolve_map.c b/src/mesa/drivers/dri/i965/intel_resolve_map.c
index bf6bcf2..5368fe3 100644
--- a/src/mesa/drivers/dri/i965/intel_resolve_map.c
+++ b/src/mesa/drivers/dri/i965/intel_resolve_map.c
@@ -39,8 +39,7 @@ intel_resolve_map_set(struct exec_list *resolve_map,
 		      enum gen6_hiz_op need)
 {
    foreach_list_typed(struct intel_resolve_map, map, link, resolve_map) {
-      if (map->level == level && map->layer == layer) {
-         map->need = need;
+      if (map->level == level && map->layer == layer && map->need == need) {
 	 return;
       }
    }
-- 
2.3.1

